MinimalMVVM/ViewModels/ObservableObject.cs

```diff
@@ -1,16 +1,31 @@
-﻿using System.ComponentModel;
+﻿using System;
+using System.ComponentModel;
 
 namespace MinimalMVVM.ViewModels
 {
+    // This implements the INotifyPropertyChanged interface in a simple way to make objects provide data-binding property change notifications
+    // To use, base a class on ObservableObject, then call RaisedPropertyChangedEvent("myPropertyName") in the set method for your property
     public abstract class ObservableObject : INotifyPropertyChanged
     {
+        // Make an event handler using (pre-existing?) PropertyChangedEventHandler delegate.
+        // This defines the signature ("shape") of the functions that can be subscribed to this event.
+        // Data-Bindings will "subscribe" to this event
         public event PropertyChangedEventHandler PropertyChanged;
 
+        // A method that's inherited by any child of this class, which is fired (using get{} and set{}) whenever a property is changed in the child class
+        // This "raises the event", or runs all the functions "subscribed", e.g. all the functions listed in the PropertyChanged event handler
         protected void RaisePropertyChangedEvent(string propertyName)
         {
+            // Set handler to the PropertyChanged event
```

Comment:
over-commented?

Reply:
Yeah, you're right, that wasn't a great comment. I think an explanation of what is going on there is useful (if only to reinforce the event terminology like "handler" for a newbie) but that comment was not any less cryptic than just reading the code. I've revised it, I think it's a little more useful now.